cdap-docs/admin-manual/source/security/authorization.rst

```diff
@@ -304,90 +312,157 @@ Streams
 
+.. _security-authorization-policies-principal:
 
-.. _security-bootstrapping-authorization:
+Kerberos Principal
+------------------
 
-Bootstrapping Authorization
-===========================
-When CDAP is first started with authorization enabled, no users are granted privileges on
-any CDAP entities. Without any privileges, it can be impossible to bootstrap CDAP (create
-a new namespace or create entities in the *default* namespace) unless an
-external interface (such as `Hue <http://gethue.com/>`__) is used for a supported
-authorization extension (such as :ref:`Integrations: Apache Sentry <apache-sentry>`).
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Operation
+     - Privileges Required
+   * - Deploy an app to impersonate a kerberos principal
+     - *ADMIN* on the principal
+   * - Create a namespace with owner prinicpal
+     - *ADMIN* on the principal
+   * - Create a dataset with owner prinicpal
+     - *ADMIN* on the principal
+   * - Create a stream with owner prinicpal
+     - *ADMIN* on the principal
 
-To make this bootstrap process easier, during startup the CDAP Master issues these grants to
-select users:
 
-- The user that runs the CDAP Master is granted *ADMIN* on the CDAP instance, so that the
-  *default* namespace can be created by that user if it does not already exist.
+.. _security-pre-grant-wildcard-privilege:
+
+Pre-grant and Wildcard Privileges
+=================================
+The new authorization model requires pre-granted privilege on all entity for any operation.
+When CDAP is first started with authorization enabled, no users are granted privileges on
+any CDAP entities. Without any privileges, CDAP will not be able to create the default namespace.
+To create the default namespace, grant *ADMIN* on default namespace to the CDAP master user.
+The default namespace will get created in several minutes automatically.
+
+To pre-grant the privilege, wildcard can be used to minimize the burden of granting privileges on all entities.
+Detailed ways of granting privileges can be found in the following sections for different authorization backends.
 
-- The user that runs the CDAP Master is granted *READ*, *WRITE*, *EXECUTE*, and
-  *ADMIN* on the system namespace, so operations such as creation of system tables,
-  deployment of system artifacts, and deployment of system dataset modules can be
-  performed by that user.
+.. _security-sentry-integration:
 
-- Additionally, a comma-separated list of users specified as the
-  ``security.authorization.admin.users`` in ``cdap-site.xml`` is granted *ADMIN*
-  privileges on the CDAP instance and the *default* namespace, so that they have the
-  required privileges to create namespaces and grant other users access to the *default*
-  namespace. It is recommended that this property be set to the list of users that will
-  administer and manage the CDAP installation.
+Sentry Integration
+------------------
```

Comment:
After this section can you add a section for Ranger and just say for ranger refer this and link to https://github.com/caskdata/cdap-security-extn/wiki/CDAP-Ranger-Extension